lib/improver/spotdata/times.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Plugins written for the Improver site specific process chain."""
+import re
+
+from datetime import datetime as dt
+from datetime import time
+from datetime import timedelta
+
+
+def get_forecast_times(forecast_date=None, forecast_time=None,
+                       forecast_length=None):
+    """
+    Generate a list of python datetime objects specifying the desired forecast
+    times. This list will be created from input specifications if provided.
+    Otherwise defaults are to start today at the most recent 6-hourly interval
+    (00, 06, 12, 18) and to run out to T+144 hours.
+
+    forecast_date : string (YYYYMMDD)
+        A string of format YYYYMMDD defining the start date for which forecasts
+        are required.
+
+    forecast_time : integer
+        An integer giving the hour on the forecast_date at which to start the
+        forecast output; 24hr clock such that 17 = 17Z for example.
+
+    forecast_length : integer
+        An integer giving the desired length of the forecast output in hours
+        (e.g. 48 for a two day forecast period).
+
```

Comment:
This should not be hard coded. I envisage this will be a suite variable that can be pulled in, but it doesn't exist yet.